Runtime/Core/ProBuilderMeshFunction.cs

```diff
@@ -51,6 +51,15 @@ void Awake()
                 Rebuild();
         }
 
+#if UNITY_EDITOR
+        void Reset()
+        {
+            // Prevent reset from Unity Editor as it would put this component in a weird state.
+            if (meshSyncState != MeshSyncState.Null)
+                UnityEditor.Undo.PerformUndo();
```

Comment:
`MeshSyncState.Null` just means that the `MeshFilter.sharedMesh` is not set, not that the `ProBuilderMesh` data is invalid. In this case I think just preventing reset entirely is the way.

Reply:
Reset gets called between Awake and Start when you generate an object. Is there something I can use to check if the data is valid?